{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "a27e490e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scanpy as sc\n",
    "\n",
    "from scipy.stats import zscore\n",
    "from scipy.stats import spearmanr\n",
    "\n",
    "import mygene"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e0f839b",
   "metadata": {},
   "source": [
    "# Spearman coefficients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "631ae871",
   "metadata": {},
   "outputs": [],
   "source": [
    "def spearman_subprocess(hallmark_df_path, adata, name_of_hallmark, species):\n",
    "    hallmark_df = pd.read_csv(hallmark_df_path)\n",
    "    sub_hallmark = hallmark_df['Subprocess'].unique()\n",
    "    sub_hallmark = list(filter(pd.notna, sub_hallmark))\n",
    "    \n",
    "    spearman_df = pd.DataFrame(columns=['Hallmark', 'Subprocess', 'Spearman', 'p-value', \n",
    "                                            'Experiment', 'Species'])\n",
    "\n",
    "    for proc in sub_hallmark:\n",
    "#         subproc_genes = list(hallmark_df[hallmark_df['Subprocess'] == f'{proc}']['ENSEMBL'])\n",
    "        subproc_genes = list(hallmark_df[hallmark_df['Subprocess'] == f'{proc}']['Gene'])\n",
    "        \n",
    "        if species == 'Mouse':\n",
    "            subproc_genes = [gene.capitalize() for gene in subproc_genes]\n",
    "            \n",
    "#         print(subproc_genes)\n",
    "            \n",
    "#         adata_process_i = adata[:, adata.var_names.isin(subproc_genes)].copy()\n",
    "        adata_process_i = adata[:, adata.var['gene_symbol'].isin(subproc_genes)].copy()\n",
    "        raw_matrix = adata_process_i.X.A\n",
    "        raw_matrix = zscore(raw_matrix, axis=0, nan_policy='omit')\n",
    "        zscored_df = pd.DataFrame(\n",
    "            raw_matrix,\n",
    "            index=adata_process_i.obs['timepoint'],\n",
    "            columns=adata_process_i.var_names\n",
    "        )\n",
    "    \n",
    "        zscored_df_long = zscored_df.reset_index().melt(\n",
    "            id_vars=zscored_df.index.name, \n",
    "            var_name='Gene', \n",
    "            value_name='Zscore'\n",
    "        )\n",
    "\n",
    "        zscored_df_long = zscored_df_long[(zscored_df_long['Zscore'] <=5) & (zscored_df_long['Zscore']>=-5)]\n",
    "    \n",
    "        zscored_df_long['timepoint_numeric'] = zscored_df_long['timepoint'].astype('category').cat.codes\n",
    "        \n",
    "        \n",
    "        rho, pval = spearmanr(zscored_df_long['timepoint_numeric'], zscored_df_long['Zscore'])\n",
    "        \n",
    "        new_row = [[name_of_hallmark, proc, rho, pval, 'Embryo', species]]\n",
    "        \n",
    "        spearman_results_df = pd.DataFrame(new_row, \n",
    "                                           columns=['Hallmark', 'Subprocess', 'Spearman', 'p-value', \n",
    "                                            'Experiment', 'Species'])\n",
    "        \n",
    "        spearman_df = pd.concat([spearman_df, spearman_results_df], ignore_index=True)\n",
    "        \n",
    "    return spearman_df\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f0f1905e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def spearman_gene(hallmark_df_path, adata, name_of_hallmark, species, sp_boarder = 0.4):\n",
    "    hallmark_df = pd.read_csv(hallmark_df_path)\n",
    "    sub_hallmark = hallmark_df['Subprocess'].unique()\n",
    "    sub_hallmark = list(filter(pd.notna, sub_hallmark))\n",
    "    \n",
    "    spearman_df = pd.DataFrame(columns=['Hallmark', 'Subprocess', 'Gene Symbol', 'Spearman', 'p-value', \n",
    "                                            'Experiment', 'Species'])\n",
    "\n",
    "    for proc in sub_hallmark:\n",
    "#         subproc_genes = list(hallmark_df[hallmark_df['Subprocess'] == f'{proc}']['ENSEMBL'])\n",
    "        subproc_genes = list(hallmark_df[hallmark_df['Subprocess'] == f'{proc}']['Gene'])\n",
    "    \n",
    "        if species == 'Mouse':\n",
    "            subproc_genes = [gene.capitalize() for gene in subproc_genes]\n",
    "            \n",
    "        \n",
    "#         adata_process_i = adata[:, adata.var_names.isin(subproc_genes)].copy()\n",
    "        adata_process_i = adata[:, adata.var['gene_symbol'].isin(subproc_genes)].copy()\n",
    "        raw_matrix = adata_process_i.X.A\n",
    "        raw_matrix = zscore(raw_matrix, axis=0, nan_policy='omit')\n",
    "        zscored_df = pd.DataFrame(\n",
    "            raw_matrix,\n",
    "            index=adata_process_i.obs['timepoint'],\n",
    "#             columns=adata_process_i.var_names\n",
    "            columns=adata_process_i.var['gene_symbol']\n",
    "        )\n",
    "    \n",
    "        zscored_df_long = zscored_df.reset_index().melt(\n",
    "            id_vars=zscored_df.index.name, \n",
    "            var_name='Gene', \n",
    "            value_name='Zscore'\n",
    "        )\n",
    "\n",
    "        zscored_df_long = zscored_df_long[(zscored_df_long['Zscore'] <=5) & (zscored_df_long['Zscore']>=-5)]\n",
    "        \n",
    "        zscored_df_long['timepoint_numeric'] = zscored_df_long['timepoint'].astype('category').cat.codes\n",
    "        \n",
    "        spearman_results = []\n",
    "        \n",
    "        for gene, group in zscored_df_long.groupby('Gene'):\n",
    "            rho, pval = spearmanr(group['timepoint_numeric'], group['Zscore'])\n",
    "            spearman_results.append((name_of_hallmark, proc, gene, rho, pval, 'Embryo', species))\n",
    "            \n",
    "        spearman_results_df = pd.DataFrame(spearman_results, \n",
    "                                           columns=['Hallmark', 'Subprocess', 'Gene Symbol', 'Spearman', \n",
    "                                                    'p-value', 'Experiment', 'Species'])\n",
    "        spearman_df = pd.concat([spearman_df, spearman_results_df], ignore_index=True)\n",
    "        \n",
    "        spearman_df_signif = spearman_df[abs(spearman_df['Spearman']) >= sp_boarder]\n",
    "    \n",
    "    print(f'Number of genes with abs(Spearman) >= {sp_boarder} = {len(spearman_df_signif)}')\n",
    "    return spearman_df_signif"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "317c26af",
   "metadata": {},
   "outputs": [],
   "source": [
    "def summary_genes(df_genes, species):\n",
    "    sub_hallmark = list(df_genes['Subprocess'].unique())\n",
    "    hallmarks = list(df_genes['Hallmark'].unique())\n",
    "    \n",
    "    spearman_df = pd.DataFrame(columns=['Hallmark', 'Subprocess', 'n_genes>0.2', 'n_genes<-0.2', \n",
    "                                        'n_genes>0.4', 'n_genes<-0.4', 'Experiment', 'Species'])\n",
    "    for hallmark in hallmarks:\n",
    "        for proc in sub_hallmark:\n",
    "            df_subset = df_genes[(df_genes['Hallmark'] == hallmark) & (df_genes['Subprocess'] == proc)]\n",
    "        \n",
    "            gene_up02 = len(df_subset[df_subset['Spearman'] >= 0.2]['Gene Symbol'])\n",
    "            gene_down02 = len(df_subset[df_subset['Spearman'] <= -0.2]['Gene Symbol'])\n",
    "            gene_up04 = len(df_subset[df_subset['Spearman'] >= 0.4]['Gene Symbol'])\n",
    "            gene_down04 = len(df_subset[df_subset['Spearman'] <= -0.4]['Gene Symbol'])\n",
    "        \n",
    "            new_row = [[hallmark, proc, gene_up02, gene_down02, gene_up04, gene_down04, 'Embryo', species]]\n",
    "            \n",
    "            spearman_results_df = pd.DataFrame(new_row, \n",
    "                                           columns=['Hallmark', 'Subprocess', 'n_genes>0.2', 'n_genes<-0.2', \n",
    "                                                    'n_genes>0.4', 'n_genes<-0.4', 'Experiment', 'Species'])\n",
    "            \n",
    "            spearman_df = pd.concat([spearman_df, spearman_results_df], ignore_index=True)\n",
    "            \n",
    "    return spearman_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e37aeef5",
   "metadata": {},
   "source": [
    "# Human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "c639f6e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "adata_human = sc.read_h5ad('32_human_adata.h5ad')\n",
    "adata_h_p = sc.read_h5ad('portal_human_v1.5.h5ad')\n",
    "adata_human.obs['timepoint'] = adata_h_p.obs['timepoint']\n",
    "\n",
    "desired_order = ['Oocyte', 'Pronucleus', 'Zygote', '2C', '4C', '8C', 'E3.0','Morula', \n",
    "                 'E4.0', 'E5.0', 'E6.0', 'E7.0', 'E8.0', 'E9.0', 'E10.0']\n",
    "\n",
    "adata_human.obs['timepoint'] = adata_human.obs['timepoint'].cat.reorder_categories(desired_order, ordered=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "752c3826",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ERX3015937_ERX3015937', 'ERX3015939_ERX3015939',\n",
       "       'ERX3015940_ERX3015940', 'ERX3015941_ERX3015941',\n",
       "       'ERX3015936_ERX3015936', 'ERX3015943_ERX3015943',\n",
       "       'ERX3015935_ERX3015935', 'ERX3015945_ERX3015945',\n",
       "       'ERX3015938_ERX3015938', 'ERX3015947_ERX3015947',\n",
       "       ...\n",
       "       'SRX300887_SRX300887', 'SRX300880_SRX300880', 'SRX300893_SRX300893',\n",
       "       'SRX300878_SRX300878', 'SRX300888_SRX300888', 'SRX300884_SRX300884',\n",
       "       'SRX300876_SRX300876', 'SRX300882_SRX300882', 'SRX300886_SRX300886',\n",
       "       'SRX300877_SRX300877'],\n",
       "      dtype='object', name='index', length=2323)"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_human.obs.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "3a326c7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_counts = pd.DataFrame(adata_human.X.A, \n",
    "             index=adata_human.obs.index,\n",
    "             columns = adata_human.var.index).T\n",
    "\n",
    "embryo_counts.to_csv(\"embryo_counts_human.csv\")\n",
    "\n",
    "embryo_meta = adata_human.obs\n",
    "embryo_meta.to_csv(\"embryo_meta_human.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "9337f879",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bec17b87",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Input sequence provided is already in string format. No operation performed\n",
      "Input sequence provided is already in string format. No operation performed\n",
      "36 input query terms found dup hits:\t[('ENSG00000276241', 4), ('ENSG00000287326', 2), ('ENSG00000243620', 2), ('ENSG00000284116', 2), ('E\n",
      "1223 input query terms found no hit:\t['ENSG00000286016', 'ENSG00000272515', 'ENSG00000290903', 'ENSG00000232597', 'ENSG00000261039', 'ENS\n"
     ]
    }
   ],
   "source": [
    "mg = mygene.MyGeneInfo()\n",
    "\n",
    "# Get your Ensembl IDs and ensure uniqueness\n",
    "ensembl_ids = list(set(adata_human.var.index.tolist()))\n",
    "\n",
    "# Query MyGene.info for gene symbols\n",
    "gene_info = mg.querymany(ensembl_ids, scopes='ensembl.gene', fields='symbol', species='human')\n",
    "\n",
    "# Convert results to a DataFrame for easier handling\n",
    "gene_info_df = pd.DataFrame(gene_info)\n",
    "\n",
    "# Drop duplicates to ensure a unique mapping\n",
    "gene_info_df = gene_info_df.drop_duplicates(subset=['query'])\n",
    "\n",
    "# Map Ensembl IDs to gene symbols\n",
    "symbol_mapping = gene_info_df.set_index('query')['symbol']\n",
    "\n",
    "# Map gene symbols to your original var index (with duplicates)\n",
    "adata_human.var['gene_symbol'] = adata_human.var.index.to_series().map(symbol_mapping).fillna(\"Unknown\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e643e31",
   "metadata": {},
   "source": [
    "## Human: subprocesses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "df0851af",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/anndata/_core/aligned_df.py:68: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n",
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/anndata/_core/aligned_df.py:68: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n"
     ]
    }
   ],
   "source": [
    "ginst_human_subp = spearman_subprocess(hallmark_df_path='genomic_signatures_new.csv', \n",
    "                                             adata = adata_human,\n",
    "                                            name_of_hallmark='Genomic Instability',\n",
    "                                            species = 'Human')\n",
    "\n",
    "sen_human_subp = spearman_subprocess(hallmark_df_path='senescence_data.csv', \n",
    "                                             adata = adata_human,\n",
    "                                            name_of_hallmark='Senescence',\n",
    "                                            species = 'Human')\n",
    "\n",
    "autoph_human_subp = spearman_subprocess(hallmark_df_path='autophagy_signatures.csv', \n",
    "                                             adata = adata_human,\n",
    "                                            name_of_hallmark='Autophagy',\n",
    "                                               species = 'Human')\n",
    "\n",
    "prot_human_subp = spearman_subprocess(hallmark_df_path='proteostasis_signatures.csv', \n",
    "                                             adata = adata_human,\n",
    "                                            name_of_hallmark='Proteostasis',\n",
    "                                             species = 'Human')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e1e415e",
   "metadata": {},
   "source": [
    "## Human: per gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "582b3b57",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes with abs(Spearman) >= 0 = 423\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n",
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes with abs(Spearman) >= 0 = 768\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n",
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n",
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n",
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n",
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:5445: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(warn_msg))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes with abs(Spearman) >= 0 = 3369\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/anndata/_core/aligned_df.py:68: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n",
      "/Users/annaponomareva/anaconda3/lib/python3.11/site-packages/anndata/_core/aligned_df.py:68: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes with abs(Spearman) >= 0 = 152\n"
     ]
    }
   ],
   "source": [
    "sen_human_genes = spearman_gene(hallmark_df_path = 'senescence_data.csv', \n",
    "                                   adata=adata_human, \n",
    "                                   name_of_hallmark='Senescence', sp_boarder = 0,\n",
    "                                   species = 'Human')\n",
    "\n",
    "autoph_human_genes = spearman_gene(hallmark_df_path='autophagy_signatures.csv', \n",
    "                                             adata = adata_human,\n",
    "                                            name_of_hallmark='Autophagy', sp_boarder = 0,\n",
    "                                          species = 'Human')\n",
    "\n",
    "prot_human_genes = spearman_gene(hallmark_df_path='proteostasis_signatures.csv', \n",
    "                                             adata = adata_human,\n",
    "                                            name_of_hallmark='Proteostasis', sp_boarder = 0,\n",
    "                                        species = 'Human')\n",
    "\n",
    "ginst_human_genes = spearman_gene(hallmark_df_path='genomic_signatures_new.csv', \n",
    "                                             adata = adata_human,\n",
    "                                            name_of_hallmark='Genomic Instability', sp_boarder = 0,\n",
    "                                         species = 'Human')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4d9109c",
   "metadata": {},
   "source": [
    "# Mouse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "b61e61bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "adata_mouse = sc.read_h5ad('01_mouse_reprocessed.h5ad')\n",
    "adata_m_p = sc.read_h5ad('portal_mouse_v1.5.h5ad')\n",
    "adata_mouse.obs['timepoint'] = adata_m_p.obs['timepoint']\n",
    "\n",
    "desired_order = ['Zygote', '2C', '4C', '8C', '16C', \n",
    "                 'E3.25', 'E3.5', 'E3.75', 'E4.5']\n",
    "\n",
    "adata_mouse.obs['timepoint'] = adata_mouse.obs['timepoint'].cat.reorder_categories(desired_order, ordered=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "1a5fd041",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_counts_mouse = pd.DataFrame(adata_mouse.X.A, \n",
    "             index=adata_mouse.obs.index,\n",
    "             columns = adata_mouse.var.index).T\n",
    "embryo_counts_mouse.to_csv(\"embryo_counts_mouse.csv\")\n",
    "\n",
    "embryo_meta_mouse = adata_mouse.obs\n",
    "embryo_meta_mouse.to_csv(\"embryo_meta_mouse.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "865e70b2",
   "metadata": {},
   "source": [
    "## Mouse: subprocesses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "50405ff9",
   "metadata": {},
   "outputs": [],
   "source": [
    "ginst_mouse_subp = spearman_subprocess(hallmark_df_path='genomic_signatures_new.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Genomic Instability',\n",
    "                                            species = 'Mouse')\n",
    "\n",
    "sen_mouse_subp = spearman_subprocess(hallmark_df_path='senescence_data.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Senescence',\n",
    "                                            species = 'Mouse')\n",
    "\n",
    "autoph_mouse_subp = spearman_subprocess(hallmark_df_path='autophagy_signatures.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Autophagy',\n",
    "                                               species = 'Mouse')\n",
    "\n",
    "prot_mouse_subp = spearman_subprocess(hallmark_df_path='proteostasis_signatures.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Proteostasis',\n",
    "                                             species = 'Mouse')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "388dfec9",
   "metadata": {},
   "source": [
    "## Mouse: per gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f665f3c3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes with abs(Spearman) >= 0 = 193\n",
      "Number of genes with abs(Spearman) >= 0 = 382\n",
      "Number of genes with abs(Spearman) >= 0 = 725\n",
      "Number of genes with abs(Spearman) >= 0 = 3049\n"
     ]
    }
   ],
   "source": [
    "ginst_mouse_genes = spearman_gene(hallmark_df_path='genomic_signatures_new.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Genomic Instability',\n",
    "                                            species = 'Mouse', sp_boarder = 0)\n",
    "\n",
    "sen_mouse_genes = spearman_gene(hallmark_df_path='senescence_data.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Senescence',\n",
    "                                            species = 'Mouse', sp_boarder = 0)\n",
    "\n",
    "autoph_mouse_genes = spearman_gene(hallmark_df_path='autophagy_signatures.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Autophagy',\n",
    "                                               species = 'Mouse', sp_boarder = 0)\n",
    "\n",
    "prot_mouse_genes = spearman_gene(hallmark_df_path='proteostasis_signatures.csv', \n",
    "                                             adata = adata_mouse,\n",
    "                                            name_of_hallmark='Proteostasis',\n",
    "                                             species = 'Mouse', sp_boarder = 0)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a83cebe",
   "metadata": {},
   "source": [
    "# Summary tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "d62fca1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_subprocess = pd.concat([sen_human_subp, autoph_human_subp, prot_human_subp, ginst_human_subp,\n",
    "                               sen_mouse_subp, autoph_mouse_subp, prot_mouse_subp, ginst_mouse_subp], \n",
    "                               ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "d8a4054a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hallmark</th>\n",
       "      <th>Subprocess</th>\n",
       "      <th>Spearman</th>\n",
       "      <th>p-value</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>DNA damage response</td>\n",
       "      <td>0.086816</td>\n",
       "      <td>7.065056e-44</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>SASP</td>\n",
       "      <td>0.075348</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>SASP_Secreted</td>\n",
       "      <td>0.003562</td>\n",
       "      <td>1.210633e-01</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>Metabolic adaptations</td>\n",
       "      <td>0.090075</td>\n",
       "      <td>2.222326e-26</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>Cell cycle arrest</td>\n",
       "      <td>0.089405</td>\n",
       "      <td>8.928841e-83</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Homologous Recombination</td>\n",
       "      <td>-0.233704</td>\n",
       "      <td>3.425171e-295</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Transcription-Coupled Repair</td>\n",
       "      <td>-0.001913</td>\n",
       "      <td>8.340411e-01</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Direct Reversal Repair</td>\n",
       "      <td>-0.123502</td>\n",
       "      <td>7.369909e-22</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Single-Strand Break Repair</td>\n",
       "      <td>-0.102632</td>\n",
       "      <td>7.094413e-25</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Exonuclease</td>\n",
       "      <td>-0.051306</td>\n",
       "      <td>2.192520e-02</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>116 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Hallmark                    Subprocess  Spearman  \\\n",
       "0             Senescence           DNA damage response  0.086816   \n",
       "1             Senescence                          SASP  0.075348   \n",
       "2             Senescence                 SASP_Secreted  0.003562   \n",
       "3             Senescence         Metabolic adaptations  0.090075   \n",
       "4             Senescence             Cell cycle arrest  0.089405   \n",
       "..                   ...                           ...       ...   \n",
       "111  Genomic Instability      Homologous Recombination -0.233704   \n",
       "112  Genomic Instability  Transcription-Coupled Repair -0.001913   \n",
       "113  Genomic Instability        Direct Reversal Repair -0.123502   \n",
       "114  Genomic Instability    Single-Strand Break Repair -0.102632   \n",
       "115  Genomic Instability                   Exonuclease -0.051306   \n",
       "\n",
       "           p-value Experiment Species  \n",
       "0     7.065056e-44     Embryo   Human  \n",
       "1     0.000000e+00     Embryo   Human  \n",
       "2     1.210633e-01     Embryo   Human  \n",
       "3     2.222326e-26     Embryo   Human  \n",
       "4     8.928841e-83     Embryo   Human  \n",
       "..             ...        ...     ...  \n",
       "111  3.425171e-295     Embryo   Mouse  \n",
       "112   8.340411e-01     Embryo   Mouse  \n",
       "113   7.369909e-22     Embryo   Mouse  \n",
       "114   7.094413e-25     Embryo   Mouse  \n",
       "115   2.192520e-02     Embryo   Mouse  \n",
       "\n",
       "[116 rows x 6 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "embryo_subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "636df87b",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_subprocess.to_csv(\"embryo_subprocess.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "81dfba4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_gene = pd.concat([sen_human_genes, autoph_human_genes, prot_human_genes, ginst_human_genes,\n",
    "                         sen_mouse_genes, autoph_mouse_genes, prot_mouse_genes, ginst_mouse_genes], \n",
    "                         ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "54d50587",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hallmark</th>\n",
       "      <th>Subprocess</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Spearman</th>\n",
       "      <th>p-value</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>DNA damage response</td>\n",
       "      <td>BAX</td>\n",
       "      <td>0.285208</td>\n",
       "      <td>1.045573e-44</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>DNA damage response</td>\n",
       "      <td>CCAR2</td>\n",
       "      <td>0.089200</td>\n",
       "      <td>1.673098e-05</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>DNA damage response</td>\n",
       "      <td>ERCC1</td>\n",
       "      <td>-0.346577</td>\n",
       "      <td>1.974057e-66</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>DNA damage response</td>\n",
       "      <td>ERCC4</td>\n",
       "      <td>-0.022108</td>\n",
       "      <td>2.878714e-01</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>DNA damage response</td>\n",
       "      <td>H2AX</td>\n",
       "      <td>0.072243</td>\n",
       "      <td>4.942496e-04</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9056</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Single-Strand Break Repair</td>\n",
       "      <td>Parp1</td>\n",
       "      <td>0.000252</td>\n",
       "      <td>9.910146e-01</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9057</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Single-Strand Break Repair</td>\n",
       "      <td>Parp2</td>\n",
       "      <td>-0.306012</td>\n",
       "      <td>1.046434e-44</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9058</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Single-Strand Break Repair</td>\n",
       "      <td>Pnkp</td>\n",
       "      <td>-0.040617</td>\n",
       "      <td>6.908370e-02</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9059</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Single-Strand Break Repair</td>\n",
       "      <td>Tdp1</td>\n",
       "      <td>0.002271</td>\n",
       "      <td>9.190791e-01</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9060</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Exonuclease</td>\n",
       "      <td>Trex1</td>\n",
       "      <td>-0.051306</td>\n",
       "      <td>2.192520e-02</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9061 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Hallmark                  Subprocess Gene Symbol  Spearman  \\\n",
       "0              Senescence         DNA damage response         BAX  0.285208   \n",
       "1              Senescence         DNA damage response       CCAR2  0.089200   \n",
       "2              Senescence         DNA damage response       ERCC1 -0.346577   \n",
       "3              Senescence         DNA damage response       ERCC4 -0.022108   \n",
       "4              Senescence         DNA damage response        H2AX  0.072243   \n",
       "...                   ...                         ...         ...       ...   \n",
       "9056  Genomic Instability  Single-Strand Break Repair       Parp1  0.000252   \n",
       "9057  Genomic Instability  Single-Strand Break Repair       Parp2 -0.306012   \n",
       "9058  Genomic Instability  Single-Strand Break Repair        Pnkp -0.040617   \n",
       "9059  Genomic Instability  Single-Strand Break Repair        Tdp1  0.002271   \n",
       "9060  Genomic Instability                 Exonuclease       Trex1 -0.051306   \n",
       "\n",
       "           p-value Experiment Species  \n",
       "0     1.045573e-44     Embryo   Human  \n",
       "1     1.673098e-05     Embryo   Human  \n",
       "2     1.974057e-66     Embryo   Human  \n",
       "3     2.878714e-01     Embryo   Human  \n",
       "4     4.942496e-04     Embryo   Human  \n",
       "...            ...        ...     ...  \n",
       "9056  9.910146e-01     Embryo   Mouse  \n",
       "9057  1.046434e-44     Embryo   Mouse  \n",
       "9058  6.908370e-02     Embryo   Mouse  \n",
       "9059  9.190791e-01     Embryo   Mouse  \n",
       "9060  2.192520e-02     Embryo   Mouse  \n",
       "\n",
       "[9061 rows x 7 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "embryo_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "56d21bad",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_gene.to_csv(\"embryo_gene.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "6ba6e273",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_summary_human_sen = summary_genes(sen_human_genes, species='Human')\n",
    "embryo_summary_human_auto = summary_genes(autoph_human_genes, species='Human')\n",
    "embryo_summary_human_prot = summary_genes(prot_human_genes, species='Human')\n",
    "embryo_summary_human_ginst = summary_genes(ginst_human_genes, species='Human')\n",
    "\n",
    "embryo_summary_mouse_sen = summary_genes(sen_mouse_genes, species='Mouse')\n",
    "embryo_summary_mouse_auto = summary_genes(autoph_mouse_genes, species='Mouse')\n",
    "embryo_summary_mouse_prot = summary_genes(prot_mouse_genes, species='Mouse')\n",
    "embryo_summary_mouse_ginst = summary_genes(ginst_mouse_genes, species='Mouse')\n",
    "\n",
    "\n",
    "\n",
    "embryo_summary = pd.concat([embryo_summary_human_sen, embryo_summary_human_auto,\n",
    "                            embryo_summary_human_prot, embryo_summary_human_ginst,\n",
    "                            embryo_summary_mouse_sen, embryo_summary_mouse_auto,\n",
    "                            embryo_summary_mouse_prot, embryo_summary_mouse_ginst], ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "75b2b49d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hallmark</th>\n",
       "      <th>Subprocess</th>\n",
       "      <th>n_genes&gt;0.2</th>\n",
       "      <th>n_genes&lt;-0.2</th>\n",
       "      <th>n_genes&gt;0.4</th>\n",
       "      <th>n_genes&lt;-0.4</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>DNA damage response</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>SASP</td>\n",
       "      <td>26</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>SASP_Secreted</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>Metabolic adaptations</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Senescence</td>\n",
       "      <td>Cell cycle arrest</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Homologous Recombination</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Transcription-Coupled Repair</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Direct Reversal Repair</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Single-Strand Break Repair</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>Genomic Instability</td>\n",
       "      <td>Exonuclease</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Embryo</td>\n",
       "      <td>Mouse</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Hallmark                    Subprocess n_genes>0.2  \\\n",
       "0             Senescence           DNA damage response           3   \n",
       "1             Senescence                          SASP          26   \n",
       "2             Senescence                 SASP_Secreted           8   \n",
       "3             Senescence         Metabolic adaptations           1   \n",
       "4             Senescence             Cell cycle arrest           5   \n",
       "..                   ...                           ...         ...   \n",
       "109  Genomic Instability      Homologous Recombination           0   \n",
       "110  Genomic Instability  Transcription-Coupled Repair           1   \n",
       "111  Genomic Instability        Direct Reversal Repair           0   \n",
       "112  Genomic Instability    Single-Strand Break Repair           0   \n",
       "113  Genomic Instability                   Exonuclease           0   \n",
       "\n",
       "    n_genes<-0.2 n_genes>0.4 n_genes<-0.4 Experiment Species  \n",
       "0              2           1            0     Embryo   Human  \n",
       "1             11          11            1     Embryo   Human  \n",
       "2             11           2            1     Embryo   Human  \n",
       "3              1           1            0     Embryo   Human  \n",
       "4              1           1            1     Embryo   Human  \n",
       "..           ...         ...          ...        ...     ...  \n",
       "109            8           0            4     Embryo   Mouse  \n",
       "110            1           0            0     Embryo   Mouse  \n",
       "111            1           0            0     Embryo   Mouse  \n",
       "112            2           0            0     Embryo   Mouse  \n",
       "113            0           0            0     Embryo   Mouse  \n",
       "\n",
       "[114 rows x 8 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "embryo_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "a278caec",
   "metadata": {},
   "outputs": [],
   "source": [
    "embryo_summary.to_csv(\"embryo_summary.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
